node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/IP.java

```diff
@@ -51,48 +54,169 @@
         return 0;
     };
 
-    /** A pool of available IP addresses */
-    public static class AddressPool {
+    /** IP configuration of a node */
+    public static class Config {
+
+        public static final Config EMPTY = new Config(Set.of(), Set.of());
+
+        private final Set<String> primary;
+        private final Pool pool;
+
+        /** DO NOT USE in non-test code. Public for serialization purposes. */
+        public Config(Set<String> primary, Set<String> pool) {
+            this.primary = ImmutableSet.copyOf(Objects.requireNonNull(primary, "primary must be non-null"));
+            this.pool = new Pool(Objects.requireNonNull(pool, "pool must be non-null"));
+        }
+
+        /** The primary addresses of this. These addresses are used when communicating with the node itself */
+        public Set<String> primary() {
+            return primary;
+        }
+
+        /** Returns the IP address pool available on a node */
+        public Pool pool() {
+            return pool;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Config config = (Config) o;
+            return primary.equals(config.primary) &&
+                   pool.equals(config.pool);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(primary, pool);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("ip config primary=%s pool=%s", primary, pool.asSet());
+        }
+
+        /** Validates and returns the given addresses */
+        public static Set<String> require(Set<String> addresses) {
+            try {
+                addresses.forEach(InetAddresses::forString);
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException("A node must have at least one valid IP address", e);
```

Comment:
The error message here seems wrong, wont this be triggered if there is at least one invalid IP? I.e. _all_ IP addresses must be valid.

Reply:
Thanks. This wasn't changed in this PR, but I'll fix the message.